crates/uv-resolver/src/resolver/urls.rs

```diff
@@ -39,12 +39,37 @@ impl Urls {
 
         // Add all direct requirements and constraints. If there are any conflicts, return an error.
         for requirement in manifest.requirements(markers, dependencies) {
-            if let Some(pep508_rs::VersionOrUrl::Url(url)) = &requirement.version_or_url {
-                if let Some(previous) = urls.insert(requirement.name.clone(), url.clone()) {
-                    if !is_equal(&previous, url) {
-                        if is_same_reference(&previous, url) {
-                            debug!("Allowing {url} as a variant of {previous}");
-                        } else {
```

Comment:
Is each branch not the same here? Do we need to split this up? `UvSource::Url { url, .. } => {` and `UvSource::Path { url, .. } => {` at least look indentical.

Reply:
Merged the url and path branch